src/main/java/io/javarig/RandomInstanceGenerator.java

```diff
@@ -25,35 +25,98 @@ private synchronized <T> T generate(Type type, Consumer<CollectionGenerator> col
         return generated;
     }
 
+    /**
+     * generate a random instance of the given type
+     *
+     * @param type the type of the object to generate
+     * @param <T>  the generic type of the object to generate
+     * @return the generated object
+     * @throws InstanceGenerationException if the instance cannot be generated for some reason (class doesn't have
+     * default constructor , class have a non-public default constructor , setter cannot be invoked ... )
+     */
     public <T> T generate(Type type) throws InstanceGenerationException {
         return generate(type, ignore -> {
         });
     }
 
+    /**
+     * generate a random instance for a collection, with a fixed size
+     *
+     * @param type the type of the object to generate
+     * @param size the size of the collection to generate
+     * @param <T>  the generated object
```

Comment:
@param <T>  the generated object
   @return the generated object
those two lines imply that the return object is the one we generated!!!!